File: C++/010-summation-of-primes.cpp
#include <iostream>
#include <math.h>
using namespace std;

int main()
{
   //Function prototype
   bool isPrime(unsigned int num);

   //local variables
   unsigned long long int sum = 2;  //start with sum include 2
   unsigned int limit = 2000000;

   for (unsigned int i = 3; i < limit; i += 2)    //skip even number
   {
      if (isPrime(i))
      {
         sum += i;
      }
   }
   cout << sum << endl;
   return 0;
}

// Function definition
bool isPrime(unsigned int num)
{
   if (num < 2) 
      return false;
   if (num == 2)   //only even number that is prime
      return true;
   if (!(num % 2))  //other even numbers: reject
      return false;
   //check modulo operation for odd numbers less than sqrt of num
   for (int i = 3; i <= sqrt(num); i += 2) 
      if (!(num % i))
         return false;
   return true;
}

File: JavaScript/010-summation-of-primes.js
/*
	The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

	Find the sum of all the primes below n.
*/

function primeSummation(n) {

	let sum = 0; // Set initial sum to zero, in case a number less than 2 is passed in

	// Check all numbers from 2 up to but not including 'n'
	// If 'n' is less than 2, the loop is never entered and the initial sum '0' is returned
	for(let num = 2; num < n; num++) {

		if(isPrime(num)) sum += num;

	}

	return sum;
}

function isPrime(num) {

	const numSqrt = Math.floor(Math.sqrt(num)); // We only have to check up to the square root to know whether a number is prime

	for(let i = 2; i <= numSqrt; i++) {

		if(num % i === 0) return false; // If it's divisible by any number, then it's not a prime factor

	}

	return true; // It is a prime as it couldn't be divided by any number
}
  
let sumOfPrimes = primeSummation(2000000);
console.log('sumOfPrimes:', sumOfPrimes)